File: app/src/main/java/me/tatarka/betterfragment/sample/MyFragment.java
package me.tatarka.betterfragment.sample;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import me.tatarka.betterfragment.Fragment;
import me.tatarka.betterfragment.ViewModelProviders;

public class MyFragment extends Fragment {
    private static final String KEY_NUMBER = "number";
    private final LifecycleLogger lifecycleLogger;

    @Inject
    public MyFragment(LifecycleLogger lifecycleLogger) {
        this.lifecycleLogger = lifecycleLogger;
    }

    public Fragment withNumber(int number) {
        getArgs().putInt(KEY_NUMBER, number)
        return this;
    }

    @Override
    public void onCreate(@Nullable Bundle savedState) {
        super.onCreate(savedState);
        getLifecycle().addObserver(lifecycleLogger);
        ViewModelProviders.of(this).get(MyViewModel.class);
        setContentView(R.layout.fragment);
        TextView number = findViewById(R.id.number);
        number.setText("" + getArgs().getInt(KEY_NUMBER));
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle state) {
        super.onSaveInstanceState(state);
        Log.d("LIFECYCLE", "ON_SAVE_INSTANCE_STATE " + this);
    }
}

File: app/src/main/java/me/tatarka/betterfragment/sample/MyFragment.kt
package me.tatarka.betterfragment.sample

import android.widget.TextView
import androidx.lifecycle.get
import me.tatarka.betterfragment.app.Fragment
import me.tatarka.betterfragment.lifecycle.ViewModelProviders
import javax.inject.Inject

private const val KEY_NUMBER = "number"

open class MyFragment @Inject constructor(
    private val lifecycleLogger: LifecycleLogger,
    private val stateLogger: InstanceStateLogger
) : Fragment() {

    fun withNumber(number: Int): Fragment = apply {
        args.putInt(KEY_NUMBER, number)
    }

    override fun onCreate() {
        super.onCreate()
        lifecycle.addObserver(lifecycleLogger)
        ViewModelProviders.of(this).get<MyViewModel>()
        setContentView(R.layout.fragment)
        requireViewById<TextView>(R.id.number).text = args.getInt(KEY_NUMBER).toString()

        instanceStateStore.add("KEY", stateLogger)
    }

}
